### ISSUE TITLE
VM: add option to insert/override (custom) precompiles

### ISSUE THREAD

**Issue by jochem-brouwer** [ethereumjs/ethereumjs-monorepo#1706]
It would be nice, especially for custom chains, to insert or override precompiles in the VM. This should be done in the constructor of the VM.

**Comment by jochem-brouwer** [comment]
This might actually also go well with a slight refactor of the precompiles. We currently would not directly support different precompiles at the same address. These are all hardcoded.

**Comment by holgerd77** [comment]
If you identify useful refactorings along these examinations would be great as well, can very well also be breaking, then just do towards `develop` for the breaking releases.

Also, just to complement: not to forget the "remove" case - beside insert/override, but I guess this can be easily integrated in some coherent data structure, somehting like:

```typescript
customPrecompiles = {
  address1 (not existing): function(), // insert
  address2 (existing): function(), // override
  address3 (exisiting): undefined // remove
}
```

? Just a very first shot.

**Comment by jochem-brouwer** [comment]
Yep, that makes sense, probably will follow the format I'm using in the custom VM opcodes (#1705).

**Comment by holgerd77** [comment]
Closed by #1813
